### PR TITLE
iidm notify update on TapChangerStepImpl

### PR DESCRIPTION
**Pull request by tadam50**
Signed-off-by: Thomas ADAM <tadam@silicom.fr>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
No

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Network listeners notifications are added on TapChangerStepImpl

**What is the current behavior?** *(You can also link to an open issue here)*
TapChangerStepImpl changes are not notified to Network listeners

**What is the new behavior (if this is a feature change)?**
TapChangerStepImpl changes are notified to Network listeners

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TapChangerStepImpl.java [hunk 1]
```diff
@@ -30,12 +34,26 @@ protected TapChangerStepImpl(double rho, double r, double x, double g, double b)
...
+    protected void setPosition(int position) {
+        this.position = position;
+    }
+
+    protected void setParent(AbstractTapChanger<?, ?, ?> parent) {
+        this.parent = parent;
+    }
+
+    protected void notifyUpdate(String attribute, Object oldValue, Object newValue) {
+        this.parent.getNetwork().getListeners().notifyUpdate(this.parent.parent.getTransformer(), () -> this.parent.getTapChangerAttribute() + ".step[" + position + "]." + attribute, oldValue, newValue);
```
Comment (yichen88) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TapChangerStepImpl.java]: `this` is not necessary.
Maybe `stepHolder` or `tapChanger` is better than `parent`

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/PhaseTapChangerAdderImpl.java [hunk 2]
```diff
@@ -107,6 +107,7 @@ public PhaseTapChangerAdder endStep() {
                 throw new ValidationException(transformer, "step b is not set");
             }
             PhaseTapChangerStepImpl step = new PhaseTapChangerStepImpl(alpha, rho, r, x, g, b);
+            step.setPosition(steps.size());
```
Comment (mathbagu) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/PhaseTapChangerAdderImpl.java]: The position could be final and set from the constructor:
```
PhaseTapChangerStepImpl step = new PhaseTapChangerStepImpl(steps.size(), alpha, rho, r, x, g, b);
```

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/RatioTapChangerAdderImpl.java [hunk 3]
```diff
@@ -96,6 +96,7 @@ public RatioTapChangerAdder endStep() {
                 throw new ValidationException(parent, "step b is not set");
             }
             RatioTapChangerStepImpl step = new RatioTapChangerStepImpl(rho, r, x, g, b);
+            step.setPosition(steps.size());
```
Comment (mathbagu) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/RatioTapChangerAdderImpl.java]: Same remark here

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TapChangerStepImpl.java [hunk 4]
```diff
@@ -12,6 +12,10 @@
  */
 class TapChangerStepImpl<S extends TapChangerStepImpl<S>> {
 
+    private AbstractTapChanger<?, ?, ?> parent;
+
+    private int position;
```
Comment (mathbagu) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/TapChangerStepImpl.java]: Same remark here

**Pull request merged**